app/models/generic_object_definition.rb

```diff
@@ -49,6 +45,24 @@ def property_defined?(attr)
     property_attribute_defined?(attr) || property_association_defined?(attr) || property_method_defined?(attr)
   end
 
+  def create_object(options)
+    GenericObject.create!({:generic_object_definition => self}.merge(options))
+  end
+
+  def find_objects(options)
+    dup = options.stringify_keys!.dup
+    ar_options   = dup.extract!(*(GenericObject.column_names - ["properties"]))
+    json_options = dup.extract!(*property_attributes.keys)
+
+    unless dup.empty?
+      err_msg = _("Attributes [%{attrs}] do not exist") % {:attrs => dup.keys.join(", ")}
```

Comment:
The error message may be misleading since you only support property_attributes for now. When a search for associations for example, the field may exist but simply not searchable.